packages/snowpack/src/commands/build.ts

```diff
@@ -52,25 +49,203 @@ async function installOptimizedDependencies(
   return installResult;
 }
 
+class BuildPipelineFile {
```

Comment:
Still needs docs, but this is the meat of the refactor: each step of the build was moved from a "full batch effort" to a "single file effort". This way, when a single file was changed we could re-run the entire build on that one file.

Reply:
Very +1 for this improvement. I’d love for this to be a stepping stone to every file being handled async, and building / watching everything in parallel.